segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/bookkeeperstorage/BookKeeperStorageTest.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.bookkeeperstorage;
+
+import io.pravega.segmentstore.storage.AsyncStorageWrapper;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageNotPrimaryException;
+import io.pravega.segmentstore.storage.StorageTestBase;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperServiceRunner;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.TestUtils;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.val;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class BookKeeperStorageTest extends StorageTestBase {
+    private static final int BOOKIE_COUNT = 1;
+    private static final int THREAD_POOL_SIZE = 20;
+
+    private static final AtomicReference<BookKeeperServiceRunner> BK_SERVICE = new AtomicReference<>();
```

Comment:
Do not use static AtomicReferences. That is just an end-run around the prohibition of mutable static variables.

Reply:
Fixed.